frontend/templates/enroll-ota.html

```diff
@@ -173,38 +224,215 @@ <h1>
           });
 
         // update image src based on OS
-        const osImagePrefix = isIPhone ? "ios" : "iPadOS";
+        const osImagePrefix = platform === "ios" ? "ios" : "iPadOS";
         document.querySelector(
           ".profile-downloaded-img"
         ).src = `{{.URLPrefix}}/assets/images/${osImagePrefix}-profile-downloaded.png`;
         document.querySelector(
           ".install-profile-img"
         ).src = `{{.URLPrefix}}/assets/images/${osImagePrefix}-install-profile.png`;
+      };
+
+      const renderError = (title, description) => {
+        const template = document.getElementById("error-template");
+        if (template) {
+          const clone = template.content.cloneNode(true);
+          clone.querySelector(".error-title").textContent = title;
+          clone.querySelector(".error-description").textContent = description;
+          document.getElementById("main-content").appendChild(clone);
+        }
+      };
+
+      const getEnrollmentSecret = () => {
+        const urlParams = new URLSearchParams(window.location.search);
+        return urlParams.get("enroll_secret");
+      };
+
+      const getEnrollmentToken = async () => {
+        const enrollSecret = getEnrollmentSecret();
+
+        const response = await fetch(
+          `${ANDROID_GET_TOKEN_URL}?enroll_secret=${enrollSecret}`,
+          {
+            method: "POST",
+            headers: {
+              "Content-Type": "application/json",
+            },
+          }
+        );
+        if (!response.ok) {
+          throw new Error("Failed to get enrollment token");
+        }
+        return await response.json();
+      };
+
+      // here we render the page content
+      document.addEventListener("DOMContentLoaded", async function () {
+        const platform = getPlatform();
+
+        // quick exit if platform is unsupported
+        if (
+          platform === "unsupported" ||
+          (!ANDROID_FEATURE_ENABLED && platform === "android") // TODO: remove android feature flag
+        ) {
+          renderContent("unsupported-template");
+          return;
+        }
+
+        // check if the enroll secret is valid.
+        if (!getEnrollmentSecret()) {
+          renderError(
+            "This URL is invalid.",
+            "Enroll secret is invalid. Please contact your IT admin."
+          );
+          return;
+        }
+
+        let templateId = getTemplateIdFromPlatform(platform);
+
+        // handle android rendering
+        if (platform === "android") {
+          if (!ANDROID_MDM_ENABLED) {
+            renderError(
+              "Android MDM is turned off.",
+              "To enroll your Android device please contact your IT admin."
+            );
+            return;
+          }
+
+          // we need to get the token and then place it in the enroll link
+          try {
+            const data = await getEnrollmentToken();
+            renderContent(templateId);
+            setEnrollTokenUrl(data.token);
```

Comment:
If `token` is empty or undefined do you want to throw an error here?

Reply:
updated